tests/test_network.py

```diff
@@ -901,17 +901,78 @@ def test_get_resource_data(self, client, network_with_data):
             for ra in group.attributes:
                 all_ras.append(ra.id)
 
+        all_resource_data = client.get_all_resource_data(s.id, include_values=False)
+        for rd in all_resource_data:
+            assert rd.value is None
+            assert int(rd.resource_attr_id) in all_ras
 
-        all_resource_data = client.get_all_resource_data(s.id, include_values='Y')
-        log.info(all_resource_data[0])
+        all_resource_data = client.get_all_resource_data(s.id, include_values=True)
         for rd in all_resource_data:
+            assert rd.value is not None
             assert int(rd.resource_attr_id) in all_ras
 
-        truncated_resource_data = client.get_all_resource_data(s.id, include_values='Y', include_metadata='Y', page_start=0, page_end=1)
+
+        truncated_resource_data = client.get_all_resource_data(s.id, include_values=True, include_metadata=True, page_start=0, page_end=1)
         assert len(truncated_resource_data) == 1
 
+    def test_get_resource_data(self, client, network_with_data):
+        net = network_with_data
+        s = net.scenarios[0]
+        node = net.nodes[0]
+
+        node_ras = [a.id for a in node.attributes]
+
+        all_resource_data = client.get_resource_data('NODE', node.id, s.id, include_values=False)
+        all_node_types = []
+        for rd in all_resource_data:
+            assert rd.dataset.value is None
+            assert int(rd.resource_attr_id) in node_ras
+            all_node_types.append(rd.dataset.type)
 
+        all_node_types = list(set(all_node_types))
```

Comment:
The set is unordered so the resulting list will have different orders on different runs and only the first element is used in the tests with `all_node_types[0]`.

Not necessarily a problem but somewhat nondeterministic: has the test ever been run for type x? Would it fail if it was?

`all_node_types` here is some permutation of `["scalar", "timeseries", "dataframe"]` so maybe testing for either specific types by name or iterating all types is more thorough.